### PR TITLE
updated buidl gradle to the 7.3 version, fixing failing unit test and…

### PR DESCRIPTION
**Pull request by CarlosIbarraTGS**
… upgraded log dependency to log4j

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -2,29 +2,19 @@ group 'com.softwareonpurpose'
...
-    compile 'com.google.code.gson:gson:2.8.+'
-    compile 'ch.qos.logback:logback-core:1.2.+'
-    compile 'ch.qos.logback:logback-classic:1.2.+'
-    compile 'org.slf4j:slf4j-api:1.7.+'
-    compile files('./src/main/resources/indentmanager-1.0.5.jar')
-    testCompile 'org.testng:testng:6.14.3'
+    implementation 'org.apache.logging.log4j:log4j-core:2.17.2'
+    implementation 'com.google.code.gson:gson:2.9.0'
+    implementation files('./src/main/resources/indentmanager-1.0.5.jar')
+    implementation 'org.testng:testng:7.5'
```
Comment (craigstockton-tgs) [inline build.gradle:13]: This should use testImplementation as it solely supports unit testing.

File: src/test/java/com/softwareonpurpose/calibrator4test/CalibratorTest.java [hunk 2]
```diff
@@ -23,10 +22,10 @@ public void nodeCalibrator_fail() {
         AnObject expected = AnObject.getInstance(true, 9, "String");
         AnObject actual = AnObject.getInstance(false, 0, "Not String");
         Calibrator calibrator = AnObjectCalibrator.getInstance(expected, actual);
-        Assert.assertEquals(calibrator.calibrate(), "CALIBRATION FAILED: \r\n" +
```
Comment (craigstockton-tgs) [inline src/test/java/com/softwareonpurpose/calibrator4test/CalibratorTest.java:26]: The modification to the end-of-line characters now fails on Windows machines.  Need to investigate a System Agnostic approach to end-of-line characters, which will set them to the appropriate value based on the OS.  There is an existing method for that, I think in one of the Apache or Commons libraries, I just can't recall then name of the method.

**Review by craigstockton-tgs — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (CarlosIbarraTGS) [reply build.gradle:13]: fixed, coming in next commit

File: src/test/java/com/softwareonpurpose/calibrator4test/CalibratorTest.java (hunk 2 above)
Comment (CarlosIbarraTGS) [reply src/test/java/com/softwareonpurpose/calibrator4test/CalibratorTest.java:26]: I'll do some research

Comment (CarlosIbarraTGS) [reply src/test/java/com/softwareonpurpose/calibrator4test/CalibratorTest.java:26]: fix made.

**Comment by CarlosIbarraTGS** [comment]
Upgraded gradle to 7.3 version, updated build.gradle to match the new specifications, and switched logging library from logback to log4j.

**Pull request merged**